index.d.ts

```diff
@@ -0,0 +1,490 @@
+    | 'MK'
+    | 'MP'
+    | 'NO'
+    | 'OM'
+    | 'PK'
+    | 'PW'
+    | 'PS'
+    | 'PA'
+    | 'PG'
+    | 'PY'
+    | 'PE'
+    | 'PH'
+    | 'PN'
+    | 'PL'
+    | 'PT'
+    | 'PR'
+    | 'QA'
+    | 'RO'
+    | 'RU'
+    | 'RW'
+    | 'RE'
+    | 'WS'
+    | 'SM'
+    | 'SA'
+    | 'SN'
+    | 'RS'
+    | 'SC'
+    | 'SL'
+    | 'SG'
+    | 'SX'
+    | 'SK'
+    | 'SI'
+    | 'SB'
+    | 'SO'
+    | 'ZA'
+    | 'GS'
+    | 'KR'
+    | 'SS'
+    | 'ES'
+    | 'LK'
+    | 'BL'
+    | 'SH'
+    | 'KN'
+    | 'LC'
+    | 'MF'
+    | 'PM'
+    | 'VC'
+    | 'SD'
+    | 'SR'
+    | 'SJ'
+    | 'SE'
+    | 'CH'
+    | 'SY'
+    | 'ST'
+    | 'TW'
+    | 'TJ'
+    | 'TZ'
+    | 'TH'
+    | 'TL'
+    | 'TG'
+    | 'TK'
+    | 'TO'
+    | 'TT'
+    | 'TA'
+    | 'TN'
+    | 'TR'
+    | 'TM'
+    | 'TC'
+    | 'TV'
+    | 'UM'
+    | 'VI'
+    | 'UG'
+    | 'UA'
+    | 'AE'
+    | 'GB'
+    | 'US'
+    | 'UY'
+    | 'UZ'
+    | 'VU'
+    | 'VA'
+    | 'VE'
+    | 'VN'
+    | 'WF'
+    | 'EH'
+    | 'YE'
+    | 'ZM'
+    | 'ZW'
+    | 'AX';
+
+// source: https://developers.google.com/maps/faq#languagesupport
+type Vue2GeocoderLanguageCode =
+    | 'ja'
+    | 'sq'
+    | 'kn'
+    | 'am'
+    | 'kk'
+    | 'ar'
+    | 'km'
+    | 'hy'
+    | 'ko'
+    | 'az'
+    | 'ky'
+    | 'eu'
+    | 'lo'
+    | 'be'
+    | 'lv'
+    | 'bn'
+    | 'lt'
+    | 'bs'
+    | 'mk'
+    | 'bg'
+    | 'ms'
+    | 'my'
+    | 'ml'
+    | 'ca'
+    | 'mr'
+    | 'zh'
+    | 'mn'
+    | 'zh-CN'
+    | 'ne'
+    | 'zh-HK'
+    | 'no'
+    | 'zh-TW'
+    | 'pl'
+    | 'hr'
+    | 'pt'
+    | 'cs'
+    | 'pt-BR'
+    | 'da'
+    | 'pt-PT'
+    | 'nl'
+    | 'pa'
+    | 'en'
+    | 'ro'
+    | 'en-AU'
+    | 'ru'
+    | 'en-GB'
+    | 'sr'
+    | 'et'
+    | 'si'
+    | 'fa'
+    | 'sk'
+    | 'fi'
+    | 'sl'
+    | 'fil'
+    | 'es'
+    | 'fr'
+    | 'es-419'
+    | 'fr-CA'
+    | 'sw'
+    | 'gl'
+    | 'sv'
+    | 'ka'
+    | 'ta'
+    | 'de'
+    | 'te'
+    | 'el'
+    | 'th'
+    | 'gu'
+    | 'tr'
+    | 'iw'
+    | 'uk'
+    | 'hi'
+    | 'ur'
+    | 'hu'
+    | 'uz'
+    | 'is'
+    | 'vi'
+    | 'id'
+    | 'zu'
+    | 'it';
+
+// source: https://developers.google.com/maps/documentation/geocoding/overview#Types
+type AddressComponentType =
+    | 'street_address'
+    | 'route'
+    | 'intersection'
+    | 'political'
+    | 'country'
+    | 'administrative_area_level_1'
+    | 'administrative_area_level_2'
+    | 'administrative_area_level_3'
+    | 'administrative_area_level_4'
+    | 'administrative_area_level_5'
+    | 'colloquial_area'
+    | 'locality'
+    | 'sublocality'
+    | 'neighborhood'
+    | 'premise'
+    | 'subpremise'
+    | 'plus_code'
+    | 'postal_code'
+    | 'natural_feature'
+    | 'airport'
+    | 'park'
+    | 'point_of_interest'
+    | 'floor'
+    | 'establishment'
+    | 'landmark'
+    | 'point_of_interest'
+    | 'parking'
+    | 'post_box'
+    | 'postal_town'
+    | 'room'
+    | 'street_number'
+    | 'bus_station';
```

Comment:
Just missing two types as per the docs

> `bus_station`, `train_station` and `transit_station` indicate the location of a bus, train or public transit stop.

```suggestion
    | 'street_number'
    | 'bus_station'
    | 'train_station'
    | 'transit_station';
```